contracts/test/fork/mainnet/4.redeemQueueMigrationV1_2_1.t.sol

```diff
@@ -0,0 +1,112 @@
+//SPDX-License-Identifier: BUSL-1.1
+
+pragma solidity 0.8.20;
+
+import "forge-std/Test.sol";
+
+import "../../../src/TUPProxy.sol";
+import "../../../src/RedeemManager.1.sol";
+import "../../../src/state/redeemManager/RedeemQueue.1.sol";
+import "../../../src/state/redeemManager/RedeemQueue.2.sol";
+import "../../../src/state/redeemManager/WithdrawalStack.sol";
+import {ITransparentUpgradeableProxy} from
+    "openzeppelin-contracts/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+
+interface MockIRedeemManagerV1 {
+    function getRedeemRequestDetails(uint32 _redeemRequestId)
+        external
+        view
+        returns (RedeemQueueV1.RedeemRequest memory);
+
+    function getRedeemRequestCount() external view returns (uint256);
+}
+
+contract RedeemQueueMigrationV1_2 is Test {
+    bool internal _skip = false;
+    string internal _rpcUrl;
+
+    address internal constant REDEEM_MANAGER_MAINNET_ADDRESS = 0x080b3a41390b357Ad7e8097644d1DEDf57AD3375;
+    address internal constant REDEEM_MANAGER_MAINNET_PROXY_ADMIN_ADDRESS = 0x2fDeF0b5e87Cf840FfE46E3A5318b1d59960DfCd;
+
+    function setUp() external {
+        try vm.envString("MAINNET_FORK_RPC_URL") returns (string memory rpcUrl) {
+            _rpcUrl = rpcUrl;
+            vm.createSelectFork(rpcUrl, 20677000);
+            console.log("1.RedeemQueueMigrationV1_2.t.sol is active");
+        } catch {
+            _skip = true;
+        }
+    }
+
+    function _generateRandomAddress(uint256 length) internal view returns (address[] memory) {
+        // Generate a random 20-byte address
+        address[] memory randomAddresses = new address[](length);
+
+        // Populate the array with random addresses
+        for (uint256 i = 0; i < length; i++) {
+            randomAddresses[i] =
+                address(uint160(uint256(keccak256(abi.encodePacked(block.timestamp, block.prevrandao, i)))));
+        }
+
+        return randomAddresses;
+    }
+
+    modifier shouldSkip() {
+        if (!_skip) {
+            _;
+        }
+    }
+
+    function test_migrate_allRequestInOneCall() external shouldSkip {
```

Comment:
I think we should check all of the redeem requests rather than a few of them.

Reply:
maybe rename the test as well to make clearer what it's checking